stl/inc/xstring

```diff
@@ -2290,8 +2298,9 @@ public:
     };
     _Bxty _Bx;
 
-    size_type _Mysize = 0; // current length of string
-    size_type _Myres  = 0; // current storage reserved for string
+    // invariant: _Myres >= _Mysize, and _Myres >= _SMALL_STRING_CAPACITY
+    size_type _Mysize = 0; // current length of string (size)
+    size_type _Myres  = _SMALL_STRING_CAPACITY; // current storage reserved for string (capacity)

```

Comment:
I'm leaving `_Myres` initialized to `_SMALL_STRING_CAPACITY`, as `_String_val` begins with SSO mode. This change will make some assignments redundant: (Candidates: [(1)](https://github.com/microsoft/STL/blob/2261f7edb760eb3fe0726187c818b796dc7ea798/stl/inc/xstring#L2630) [(2)](https://github.com/microsoft/STL/blob/2261f7edb760eb3fe0726187c818b796dc7ea798/stl/inc/xstring#L2650) [(3)](https://github.com/microsoft/STL/blob/2261f7edb760eb3fe0726187c818b796dc7ea798/stl/inc/xstring#L2686); they are unchanged now)